### PR TITLE
feat: deal with more than 128 txn

### PR DESCRIPTION
**Pull request by fengjiachun**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

_PLEASE DO NOT LEAVE THIS EMPTY !!!_

Please explain IN DETAIL what the changes are in this PR and why they are needed:

etcd's default configuration cannot execute more than 128 txn at once, this PR is to solve this problem.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Review by sunng87 — APPROVED** [review]

**Review by shuiyisong — APPROVED** [review]

File: src/meta-srv/src/service/store/etcd.rs [hunk 1]
```diff
@@ -185,24 +229,20 @@ impl KvStore for EtcdStore {
...
-                        prev_kvs.push(KvPair::from_etcd_kv(prev_kv));
+        for txn_res in txn_responses {
+            for op_res in txn_res.op_responses() {
+                match op_res {
+                    TxnOpResponse::Put(put_res) => {
+                        if let Some(prev_kv) = put_res.prev_key() {
+                            prev_kvs.push(KvPair::from_etcd_kv(prev_kv));
+                        }
                     }
+                    _ => unreachable!(), // never get here
```
Comment (killme2008) [inline src/meta-srv/src/service/store/etcd.rs:244]:
```suggestion
                    _ => unreachable!(),
```

File: src/meta-srv/src/service/store/etcd.rs [hunk 2]
```diff
@@ -232,28 +272,23 @@ impl KvStore for EtcdStore {
...
-                    });
+        for txn_res in txn_responses {
+            for op_res in txn_res.op_responses() {
+                match op_res {
+                    TxnOpResponse::Delete(delete_res) => {
+                        delete_res.prev_kvs().iter().for_each(|kv| {
+                            prev_kvs.push(KvPair::from_etcd_kv(kv));
+                        });
+                    }
+                    _ => unreachable!(), // never get here
```
Comment (killme2008) [inline src/meta-srv/src/service/store/etcd.rs:286]:
```suggestion
                    _ => unreachable!(),
```

File: src/meta-srv/src/service/store/etcd.rs [hunk 3]
```diff
@@ -24,13 +24,16 @@ use common_error::prelude::*;
...
     Client, Compare, CompareOp, DeleteOptions, GetOptions, PutOptions, Txn, TxnOp, TxnOpResponse,
+    TxnResponse,
 };
 
 use crate::error;
 use crate::error::Result;
 use crate::metrics::METRIC_META_KV_REQUEST;
 use crate::service::store::kv::{KvStore, KvStoreRef};
 
+const MAX_TXN_SIZE: usize = 128;
```
[the remark this answers is not in the document]
Comment (fengjiachun) [reply src/meta-srv/src/service/store/etcd.rs:35]: fixed in #1801

File: src/meta-srv/src/service/store/etcd.rs [hunk 4]
```diff
@@ -51,6 +54,51 @@ impl EtcdStore {
...
+            let txn_res = self
+                .client
+                .kv_client()
+                .txn(txn)
+                .await
+                .context(error::EtcdFailedSnafu)?;
+            return Ok(vec![txn_res]);
+        }
+
+        let mut txns = vec![];
```
[the remark this answers is not in the document]
Comment (fengjiachun) [reply src/meta-srv/src/service/store/etcd.rs:71]: fixed in #1801

File: src/meta-srv/src/service/store/etcd.rs [hunk 5]
```diff
@@ -51,6 +54,51 @@ impl EtcdStore {
...
+            let txn_res = self
+                .client
+                .kv_client()
+                .txn(txn)
+                .await
+                .context(error::EtcdFailedSnafu)?;
+            txn_responses.push(txn_res);
+        }
+        Ok(txn_responses)
+    }
```
[the remark this answers is not in the document]
Comment (fengjiachun) [reply src/meta-srv/src/service/store/etcd.rs:101]: fixed in #1801

File: src/meta-srv/src/service/store/etcd.rs (hunk 1 above)
Comment (fengjiachun) [reply src/meta-srv/src/service/store/etcd.rs:244]: fixed in #1801

**Pull request merged**